### PR TITLE
BLOCKCHAIN-260 - Allow company owner to request unregistration

### PR DESCRIPTION
**Pull request by MichalLustykNeti**

### CODE REVIEW

File: src/api/nodeRpcCall.js [hunk 1]
```diff
@@ -1242,6 +1251,13 @@ const requestEditCompanyRegistration = async (companyData, companyId, walletAddr
   return submitExtrinsic(extrinsic, walletAddress);
 }
 
+const deleteRequestCompanyRegistration = async (companyId, walletAddress)  => {
```
Comment (kacperzuk-neti) [inline src/api/nodeRpcCall.js:1254]: Other PR adds the same thing under different name: https://github.com/liberland/liberland_frontend/pull/162/files#diff-23f1d917744ae222dc753c21f0aea86fd32bbd65ef4e552d1650bba52573b51eR2098-R2103

This seems to be out of scope for this PR though? Why is it here?

File: src/components/Offices/CompanyRegistry/index.js [hunk 2]
```diff
@@ -76,8 +79,13 @@ function CompanyRegistration({ registration }) {
...
       <div className={styles.buttonWrapper}>
-        <Button primary medium red onClick={unregister}>
-          Unregister company
+        <Button
+          primary
+          medium
+          red
+          onClick={() => unregisterCompany(registration.entity_id)}
+        >
+          Unregister company hard
```
Comment (kacperzuk-neti) [inline src/components/Offices/CompanyRegistry/index.js:88]: `soft` vs `hard` aren't very good names, let's not show it to the user. Let's leave simple `Unregister company` here.

File: src/components/Offices/CompanyRegistry/index.js [hunk 3]
```diff
@@ -53,11 +53,14 @@ function MissingRequest() {
...
+
   if (!registration) return null;
   if (!registration.registration) return <div>Company not registered yet</div>;
-  const unregister = () => {
+
+  const unregisterCompany = (entityId) => {
     dispatch(officesActions.unregisterCompany.call({
-      companyId: registration.entity_id,
+      entityId,
+      soft: true,
```
Comment (kacperzuk-neti) [inline src/components/Offices/CompanyRegistry/index.js:63]:
```suggestion
      soft: false,
```

File: src/components/Offices/CompanyRegistry/index.js [hunk 4]
```diff
@@ -141,16 +150,46 @@ function CompanyRequest({ request }) {
...
+  const dispatch = useDispatch();
+  const unregisterCompany = () => {
+    dispatch(officesActions.unregisterCompany.call({
+      entityId,
+      soft: true,
+    }));
+  };
+  return (
+    <div>
+      <p>User requested company unregistartion.</p>
```
Comment (kacperzuk-neti) [inline src/components/Offices/CompanyRegistry/index.js:174]:
```suggestion
      <p>User requested company unregistration.</p>
```

File: src/components/Offices/CompanyRegistry/index.js [hunk 5]
```diff
@@ -141,16 +150,46 @@ function CompanyRequest({ request }) {
...
+      <div className={styles.buttonWrapper}>
+        <Button
+          primary
+          medium
+          red
+          onClick={() => unregisterCompany()}
+        >
+          Unregister company soft
```
Comment (kacperzuk-neti) [inline src/components/Offices/CompanyRegistry/index.js:182]:
```suggestion
          Approve request & Unregister company
```

**Review by kacperzuk-neti — CHANGES_REQUESTED** [review]

File: src/api/nodeRpcCall.js (hunk 1 above)
Comment (MichalLustykNeti) [reply src/api/nodeRpcCall.js:1254]: I fot this is also a part of this PR

Comment (kacperzuk-neti) [reply src/api/nodeRpcCall.js:1254]: Okay, will deduplicate this when merging :+1:

**Review by kacperzuk-neti — APPROVED** [review]

File: src/api/nodeRpcCall.js (hunk 1 above)
Comment (DorianSternVukotic) [reply src/api/nodeRpcCall.js:1254]: Other PR is merged, fix please

**Review by DorianSternVukotic — CHANGES_REQUESTED** [review]

File: src/api/nodeRpcCall.js (hunk 1 above)
Comment (kacperzuk-neti) [reply src/api/nodeRpcCall.js]: Done.
